arch-design/delfin/PerfomanceMontoringDesign.md

```diff
@@ -1,314 +1,313 @@
+
 | | |
 |-|-|
 |**Resource**|**Required labels**|
-|Array| **storage_id** = <storage_id> **resource_type** = array **id** = < delfin id of this resource> |
-|Storage-Pool| **storage_id** = <storage_id> **resource_type** = storage_pool **id** = < delfin id of this resource> **native_storage_pool_id** = < associated storage_pool id in backend > **type**=enum **unit** = enum **value_type** = enum|
-|Volume|**storage_id** = <storage_id> **resource_type** = volume **id** = < delfin id of this resource> **native_volume_id** = < id of this pool  in back_end> **native_storage_pool_id** = < associated storage_pool id in backend > **type**=enum **unit** = enum **value_type** = enum|
+|Device| **storage_id** = <storage_id> <br /> **id** = < delfin id> <br />  **resource_type** = device <br /> **resource_id** = < native device id> <br /> **unit** = enum <br /> **type** = enum |
+|Storage-Pool| **storage_id** = <storage_id> <br /> **resource_type** = storage_pool <br /> **resource_id** = < native  id of this resource> <br />  **id** = < delfin id> <br />  **type**=enum <br />  **unit** = enum <br />  
+
 
-#### enums
+###### Label field enums
 | | |
 |-|-|
 |**Propperty**|**enum**|
 |type|RAW,DERIVED,AGGREGATED|
-|value_type|COUNTER, RATE, GUAGE,|
+|
 |unit|IOPS,MB/s,%,ms,KB,
-## REST interface for configuring performance collection 
-![](./Resources/PUT.PNG)
-![](./Resources/PUT_1.PNG)
-![](./Resources/PUT_2.PNG)
 
-## Driver interfaces
-```
-collection_spec = {
-'interval':900,
-'is_historic':True,
-'storage_id': '12c2d52f-01bc-41f5-b73f-7abf6f38a2a6'
-}
+##### Database model
+To be updated
+
+
+##### Metric lists
+###### Metric Lists analysis summary
+* Considered EMC SRM, stor2rrd, OCI documents and solarwinds SRM dashboard   to derive support matrix of some common back ends (EMC vmax, EMC unity, ibm ds, ibm SVC, HP 3par)
+* Some metrics can take common names in delfin
+* Other metrics will follow platform metric naming
+* Delfin will have API to list supported metric for each resource at each driver
+* Metric name bandwidth is rarely used , most of the platforms consider data rate as throughput (MB/s) and IOPS as requests or iorate 
+* Common metric names are derived based on their availability in other SRM platforms 
+
+
+###### Common metric list
+
+**Device**
+
+| Delfin Metric  | Unit | Description                                                         |
+| -------------- | ---- | ------------------------------------------------------------------ |
+| throughput     | MB/s | Represents how much data is successfully transferred in MB/s       |
+| responseTime   | ms   | Average time taken for an IO operation in ms                       |
+| requests       | IOPS | Input/output operations per second                                 |
+| readThroughput | MB/s | Represents how much data read is successfully transferred in MB/s  |
+| writeThrouput  | MB/s | Represents how much data write is successfully transferred in MB/s |
```

Comment:
Is this the average number of data written to disk? or is the exact data?

Reply:
MB/s ->Average Data tarnsfered per second .
This again based on platform description, some may give cumilative data. When there is change in the description of metric driver has to give 'native_description' in the labels.